### PR TITLE
Resolving frontend views

### LINKED ISSUE #1090: Update and move frontend_views
The usage of the `frontent_views` folder is not compatible with the Django convention.

### Tasks

- [x] Reformat `frontend_views/views.py`
    - [x] If it is possible, use f-Strings
    - [x] Fix variable names. The name of the variables should be self-explanatory.
           **Examples of bad naming:**
https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L21
https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L100

- [x] `frontend_views/views.py` -> `accounts/views.py`
     - [x] [user_profile](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L61)
     - [x] Related URLs (`frontend_views/urls.py` -> `accounts/urls/urls.py`)

 - [x] `frontend_views/views.py` -> `threads/views.py`
     - [x] [base_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L17)
     - [x] [issue_thread](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L95)
     - [x] [civi2csv](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L189)
     - [x] Related URLs (`frontend_views/urls.py` -> `threads/urls.py`)

- [x] Discuss which app should include below:
    - [x] [create_group](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L161)
    - [x] [declaration](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L165)
    - [x] [landing_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L169)
    - [x] [how_it_works_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L173)
    - [x] [about_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L177)
    - [x] [support_us_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L181)
    - [x] Except for `create_group`, it might be good to use [TemplateView](https://docs.djangoproject.com/en/3.2/ref/class-based-views/base/#templateview) for rest.
    - [x] Related URLs
    - [x] A good reference would be great to follow the Django convention about the views above. For example, `/about/` page does not belong to any model. We should decide where to put it. `thread/views.py` or a new file `core/views.py`?
  
- [x] Refactor for all the usages above. (including all `views.py` files and JS files)

- [x] Remove `frontend_views` folder

Since this is a long task and it requires some knowledge about the project. @AbhijithGanesh, I am assigning this issue.
Please start this issue after merging PR #1088 and #1089.

### PR DESCRIPTION
**Pull request by AbhijithGanesh**
Closes issue #1090

This issue resolves the following:
- Removed files related to frontend_views and the directory itself
- Fixed improper naming and gives actual literal names
- Moves [ https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py ](user_profile) and related urls
- Moves frontend views to threads

### CODE REVIEW

**Comment by AbhijithGanesh** [comment]
@brylie New PR, same issues

**Comment by brylie** [comment]
Would you please add to your pull request description a bit more information about the parts of #1090 that this pull request resolves?

**Comment by brylie** [comment]
If possible, split issue #1090 into multiple issues so this pull request can close one of the issues.

**Comment by AbhijithGanesh** [comment]
If you merge this, it should be good I guess

**Comment by AbhijithGanesh** [comment]
@brylie I have updated the description and this branch can be merged

File: project/accounts/urls/urls.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 from django.urls import path
 from django.contrib.auth import views as auth_views
 from accounts.views import (RegisterView, SettingsView, ProfileActivationView, PasswordResetView, PasswordResetDoneView,
-                            PasswordResetConfirmView, PasswordResetCompleteView, ProfileSetupView)
+                            PasswordResetConfirmView, PasswordResetCompleteView, ProfileSetupView, user_profile)
+
+from threads.views.views import base_view
 
 urlpatterns = [
+    path("", base_view),
```
Comment (gorkemarslan) [inline project/accounts/urls/urls.py]: `base_view` should not be in `accounts` app. You add it to `threads` app, as well. Please remove this and the related import.

File: project/core/urls.py [hunk 2]
```diff
@@ -19,17 +19,21 @@
...
 from threads import urls as threads
 
+
+from threads.views.views import about_view, support_us_view, how_it_works_view, base_view
 from core.router import CiviWikiRouter
 
+
 urlpatterns = [
     path("admin/", admin.site.urls),
+    path("", base_view),
```
Comment (gorkemarslan) [inline project/core/urls.py]: Remove it because you use it in `threads.urls.urls` in this branch.

File: project/core/urls.py [hunk 3]
```diff
@@ -47,7 +51,9 @@
         "mstile-144x144.png",
         RedirectView.as_view(url="/static/favicon/mstile-144x144.png"),
     ),
-    path("", include(frontend_views)),
+    path("about/", about_view, name="about"),
+    path("support_us/", support_us_view, name="support us"),
+    path("howitworks/", how_it_works_view, name="how it works"),
```
Comment (gorkemarslan) [inline project/core/urls.py]: Remove them, they are in `threads.urls.urls `in your branch.

File: project/threads/urls/urls.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+
+urlpatterns = [
+    path("about/", about_view, name="about"),
+    path("", base_view, name="base"),
+    path("support_us/", support_us_view, name="support us"),
+    path("howitworks/", how_it_works_view, name="how it works"),
+    path("declaration", declaration, name="declaration"),
+    path("create-group", create_group, name="create group"),
```
Comment (gorkemarslan) [inline project/threads/urls/urls.py]: Could you add ending slash `'/'` to them?

File: project/threads/views/views.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+    queryset = Category.objects.all()
+    serializer_class = CategorySerializer
+    authentication_classes = ()
+
+    @action(detail=True)
+    def threads(self, request, pk=None):
+        category_threads = Thread.objects.filter_by_category_id(pk)
+        serializer = ThreadSerializer(category_threads, many=True)
+        return Response(serializer.data)
```
Comment (gorkemarslan) [inline project/threads/views/views.py]: `CategoryViewSet` somehow appears in the `threads` app but it is also present in the `category` app. Could you remove this one and fix the import?

**Review by gorkemarslan — CHANGES_REQUESTED** [review]

**Comment by AbhijithGanesh** [comment]
@gorkemarslan finished those changes

File: project/accounts/urls/urls.py [hunk 6]
```diff
@@ -1,7 +1,9 @@
 from django.urls import path
 from django.contrib.auth import views as auth_views
 from accounts.views import (RegisterView, SettingsView, ProfileActivationView, PasswordResetView, PasswordResetDoneView,
-                            PasswordResetConfirmView, PasswordResetCompleteView, ProfileSetupView)
+                            PasswordResetConfirmView, PasswordResetCompleteView, ProfileSetupView, user_profile)
+
+from threads.views.views import base_view
```
Comment (gorkemarslan) [inline project/accounts/urls/urls.py]: You can remove this unused import.

**Review by gorkemarslan — CHANGES_REQUESTED** [review]

File: project/core/urls.py [hunk 7]
```diff
@@ -19,17 +19,20 @@
 from django.contrib import admin
 from django.urls import include, path
 from django.views.generic.base import RedirectView
-from frontend_views import urls as frontend_views
 from threads import urls as threads
 
+
+from threads.views.views import about_view, support_us_view, how_it_works_view, base_view
```
Comment (gorkemarslan) [inline project/core/urls.py]: You can remove these two lines including `from threads import urls as threads`.

**Review by gorkemarslan — CHANGES_REQUESTED** [review]

File: project/core/urls.py (hunk 7 above)
Comment (AbhijithGanesh) [reply project/core/urls.py]: I will remove them

Comment (AbhijithGanesh) [reply project/core/urls.py]: Removed

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, I see that you changed static file imports to be via CDN in the templates. Is there any open issue for that. Or you just want to improve code quality?

**Comment by AbhijithGanesh** [comment]
There is no open issue for that since as it deals with dependency files(which can be directly loaded from CDN)

**Comment by gorkemarslan** [comment]
It would be better if you specify your purpose on usage of CDN in the PR message.

Except for CDN, the PR is fine to me.

Let's wait for @brylie to decide to keep CDN usage or revert them back.

If @brylie says you are allowed to make these changes in CDN, then you need to remove the related files in the [dependencies](https://github.com/CiviWiki/OpenCiviWiki/tree/develop/project/core/templates/static/dependencies) folder because no need to use them anymore. In that situation, you also need to edit the the issue #1090 to specify CDN usages in order to track it in the future.

If @brylie says let's handle this situation in another issue, you should revert them.

In either case, before pushing any commits related to CDN, let's wait @brylie. :smile:  

Another point, the number of commits you pushed is a little bit excessive. When you open this PR, you've already added more than 30 commits. That seems scary in the first glance. Next time, could you squash relevant changes to be just one commit instead of many in order to decrease your number of commits? :smile:

**Comment by AbhijithGanesh** [comment]
I haven't made any changes tbh, I just reverted my commit to make sure we can add it to a different issue

**Comment by AbhijithGanesh** [comment]
I will reduce the number of commits next time

**Review by gorkemarslan — APPROVED** [review]

**Comment by AbhijithGanesh** [comment]
Also massive thanks to @gorkemarslan for going through this for the n-th time, I have virtually lost count, thank you for this wonderful work

**Comment by brylie** [comment]
> Let's wait for @brylie to decide to keep CDN usage or revert them back.
>
> If @brylie says you are allowed to make these changes in CDN, then you need to remove the related files in the dependencies folder because no need to use them anymore. In that situation, you also need to edit the the issue #1090 to specify CDN usages in order to track it in the future.

We can keep the CDN links and remove the related files, as @gorkemarslan mentioned.

**Review by brylie — CHANGES_REQUESTED** [review]
Please resolve conflicts.

**Review by brylie — APPROVED** [review]

**Comment by AbhijithGanesh** [comment]
Can you merge this soon brylie?

**Comment by brylie** [comment]
I approved these changes and will merge the pull request once the pull request is linked to the related GitHub issue via a supported keyword.

When creating a pull request that will resolve an issue, please [link the pull request to the related issue(s)](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) by using one of the [supported keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) followed by the issue number.

For example:

```
Closes #<issue-number>
```

## Further details

> You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message.
> 
> - close
> - closes
> - closed
> - fix
> - fixes
> - fixed
> - resolve
> - resolves
> - resolved

**Comment by brylie** [comment]
Note that GitHub is a bit picky with the supported keywords, so you need to follow the syntax exactly as described in the official documentation

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by brylie** [comment]
When the syntax is correct, the linked issue will appear in the pull request sidebar section depicted below:

image

**Comment by brylie** [comment]
The current comment is close, but not close enough for GitHub to recognize the link

image

**Comment by brylie** [comment]
@AbhijithGanesh, notice how the comment in pull request #1124 is automatically linked to issue #1101

image

**Comment by AbhijithGanesh** [comment]
Closes #1090

**Comment by AbhijithGanesh** [comment]
Now it shows

**Pull request merged**